handlers/pipeline.go

```diff
@@ -136,6 +136,50 @@ func PipelineGet(c echo.Context) error {
 	return c.String(http.StatusNotFound, errPipelineNotFound.Error())
 }
 
+// PipelineDelete accepts a pipeline id and deletes it from the
+// store. It also removes the binary inside the pipeline folder.
+func PipelineDelete(c echo.Context) error {
+	pipelineIDStr := c.Param("pipelineid")
+
+	pipelineID, err := strconv.Atoi(pipelineIDStr)
+	if err != nil {
+		return c.String(http.StatusBadRequest, errInvalidPipelineID.Error())
+	}
+
+	// Look up pipeline for the given id
+	var foundPipeline *gaia.Pipeline
```

Comment:
There is no need for this to be a reference since you are passing in a copy anyways. Let's keep this on the stack with removing the pointer and the address passing.

Reply:
I changed it to a reference type as per @michelvocks suggestion.
> You can make foundPipeline a pointer. Then you can check here: if foundPipeline == nil

Thoughts?

Reply:
You can check if it is initialized by comparing it to an empty struct of the same type. Rather than a pointer which might be stored on the heap.

Reply:
I guess it doesn't matter much but avoids a nil pointer expedition.

Reply:
@bidhan-a Sorry. My mistake here. You implementation was actually right. :sweat_smile: